rclpy/rclpy/context.py

```diff
@@ -68,7 +68,7 @@ def init(self,
             rclpy_implementation.rclpy_init(
                 args if args is not None else sys.argv,
                 capsule,
-                domain_id)
+                domain_id if domain_id is not None else rclpy_implementation.RCL_DEFAULT_DOMAIN_ID)
```

Comment:
@sloretz meta: I presume C++ APIs with optional arguments get tricky enough to warrant exposing `rcl` sentinel all the way up. Is that true? An optional `domain_id` would be best.

Reply:
There are at least a couple other ways besides exposing `RCL_DEFAULT_DOMAIN_ID` as a constant, but I liked it the most. I'd still be happy to change it if you'd like.

I think the way you're saying is best is to make the argument a py::object and do the conditional in C++

```c++
void
init(py::list pyargs, py::capsule pycontext, py::object pydomain_id)
{
  size_t domain_id = RCL_DEFAULT_DOMAIN_ID;
  if (!pydomain_id.is_none()) {
      domain_id = pydomain_id.cast<size_t>();
  }
//...
}
```

Another way is to make `domain_id` a keyword argument and set the default value to `RCL_DEFAULT_DOMAIN_ID`


```c++
  m.def(
    "rclpy_init", &rclpy::init,
    "Initialize RCL.",
    py::arg("args"), py::arg("context"), py::arg("domain_id") = RCL_DEFAULT_DOMAIN_ID);
```

The python code has to either know the default value or have two different calls to support it's own keyword arg because `None` can't be converted to `size_t`.

```python
        if domain_id is None:
            rclpy_implementation.rclpy_init(
                args if args is not None else sys.argv,
                capsule)
        else:
            rclpy_implementation.rclpy_init(
                args if args is not None else sys.argv,
                capsule,
                domain_id)
```

Reply:
Oh, the first looks neat! You can even skip setting a domain id entirely if one's not specified, `rcl` and `rmw` will take care of providing a suitable default.

But I won't block on this. It's up to you.

Reply:
I'll leave this one as is.